frontend/src/app/core/exercise.service.ts

```diff
@@ -251,15 +251,12 @@ export class ExerciseService {
                         !client.isInWaitingRoom
                 ),
                 switchMap((client) => this.store.select(selectVisibleVehicles)),
-                withLatestFrom(this.store.select(selectCurrentTime)),
+                // Do not trigger the message if the vehicle was removed and added again in the same millisecond
+                debounce(() => timer(1)),
```

Comment:
```suggestion
                debounceTime(0),
```

Or `1` if it didn't work with `0`. Though, I'm currently missing why it shouldn't :/

Reply:
It works that way, but I don't find it obvious that one output per 0ms (which is the meaning of debounceTime(0)), does discard any events. So I rather use 1 here because that makes it obvious that there is a real time span during which outputs are discarded.

Reply:
I would put it like this:
The 0 confuses an inexperienced JS-developer, while the 1 confuses an experienced JS-developer.

If there is a 0, I know it is just skipping multiple synchronous emits. If there is a 1, I think this is about latency and could potentially break on different systems, etc.

Reply:
Also, I realized there is one more Problem with the notification coming up when the viewport one is restricted to Changed, which also should not happen

Reply:
> I would put it like this: The 0 confuses an inexperienced JS-developer, while the 1 confuses an experienced JS-developer.
> 
> If there is a 0, I know it is just skipping multiple synchronous emits. If there is a 1, I think this is about latency and could potentially break on different systems, etc.

I just found the word timeout in the documentation confusing since it is more of a time window that closes after that time so in the time window of 0 can be multiple elements but if I wait 0 seconds for more events it sounds like there is a maximum of one event, but I think 0 is the optimal solution after reading the documentation a second time

Reply:
I believe the naming is derived from the [`setTimeout()` function](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout) that's part of the browser- and nodejs-API.

Reply:
I think I now fixed both things the movement in a simulated region issue with debounceTime and the switch viewport issue with moving pairwise inside switch map so only pairs from the same viewport are built

Reply:
I believe `pairwise` should come before `debounce `.
This way you should also be able to move them back to the outer pipe?